### DISCUSSION TITLE
custom theme is overrided by default theme

### DISCUSSION THREAD

**Discussion by zgid123** [chakra-ui/chakra-ui#8587] [Q&A] [answered]
Hi, I am building a design system using chakra-ui/react 2.8.2 in monorepo project.

I create a package for this project.  I create a provider wrapper 

```tsx
export function ThemeProvider({
  children,
  fontComponent,
  theme: themeProps = {} as Theme,
}: IThemeProviderProps): JSX.Element {
  const FontComponent = fontComponent || Fonts;
  const themeConfig = mergeThemeOverride(theme, themeProps);

  return (
    <ChakraProvider theme={themeConfig}>
      <FontComponent />
      {children}
    </ChakraProvider>
  );
}
```

Chakra-ui will insert `<style data-emotion="css-global" data-s="">` that contains root css `:host, :root, [data-theme]` twice, and my custom css is append before the default theme of chakra-ui, so my theme is overrided by the default one.

How can I fix this one?

**Comment by stevaras** [comment]
@zgid123 Hey mate, I had a similar issue and eventually had to solve it by using cssVarsRoot="#some-id-in-my-app" This made our custom theme override the default chakra one

**Comment by segunadebayo** [comment, marked as answer]
v2 inserted the default Emotion global after yours, so the default theme won. v2 is maintenance-only.

In v3, `createSystem(defaultConfig, yours)` merges your tokens on top. Pass that system to `ChakraProvider value={system}`.
